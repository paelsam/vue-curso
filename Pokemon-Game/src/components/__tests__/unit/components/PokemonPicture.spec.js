import { beforeEach, describe, expect, test } from "vitest";
import { shallowMount } from "@vue/test-utils";
import PokemonPicture from '@/components/PokemonPicture.vue'


describe('PokenmonPicture components', () => {
    
    
    test('debe hacer match con el snapshot', () => {
        const wrapper = shallowMount( PokemonPicture, {
            props: {
                pokemonId: 1,
                showPokemon: false
            }
        })
        expect( wrapper.html() ).toMatchSnapshot();
    })

    test('debe de mostrar la imagen oculta y el pokemón 100', () => {

        const wrapper = shallowMount( PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: false
            }
        })


        const [ img1, img2 ] = wrapper.findAll('img')
        expect( img1.exists() ).toBeTruthy()
        expect( img2 ).toBe(undefined)

        expect( img1.classes('hidden-pokemon') ).toBeTruthy()
        expect( img1.attributes().src )
            .toBe(`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${wrapper.props().pokemonId}.svg`)

    })

    test('debe de mostrar el pokemon si showPokemon:true', () => {
        const wrapper = shallowMount( PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: true
            }
        })

        const [ img1, img2 ] = wrapper.findAll('img')

        // Si showPokemon es verdadero la img2 pasa a ser la img1
        // Por lo tanto, img2 seguirá siendo undefined
        expect( img1.exists() ).toBeTruthy()
        expect( img2 ).toBe(undefined)

        expect( img1.classes('fade-in') ).toBeTruthy()
        expect( img1.classes('hidde-pokemon') ).toBeFalsy()
    })

})