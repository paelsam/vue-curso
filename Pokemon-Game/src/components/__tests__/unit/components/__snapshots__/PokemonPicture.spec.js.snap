// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PokenmonPicture components > debe hacer match con el snapshot 1`] = `
"<div data-v-c76ad58e=\\"\\" class=\\"pokemon-container\\"><img data-v-c76ad58e=\\"\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg\\" class=\\"hidden-pokemon\\" alt=\\"pokemon\\">
  <!--v-if-->
</div>"
`;
